### PR TITLE
Add Android View Box Implementation

### PR DESCRIPTION
**Pull request by underscoretang**
Screenshot 2023-10-12 at 2 26 10 PM

### CODE REVIEW

**Comment by underscoretang** [comment]
Current dependencies on/for this PR:
* trunk
  * **PR #1585** <a href="https://app.graphite.dev/github/pr/cashapp/redwood/1585" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/cashapp/redwood/1585?utm_source=stack-comment).

File: redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+      val height = when (layoutParams.height) {
+        LayoutParams.MATCH_PARENT -> heightMeasureSpec
+        LayoutParams.WRAP_CONTENT -> widgetChildren.maxOfOrNull { it.value.measuredHeight } ?: 0
+        else -> layoutParams.height
+      }
+
+      // Call to super here triggers a layout pass on the children.
+      super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+      setMeasuredDimension(width, height)
```
Comment (underscoretang) [inline redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt:83]: @swankjesse, I think this was the piece we were missing prior. wdyt?

File: redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      // Get width and height values from layout parameters
+      val width = when (layoutParams.width) {
+        LayoutParams.MATCH_PARENT -> widthMeasureSpec
+        LayoutParams.WRAP_CONTENT -> widgetChildren.maxOfOrNull { it.value.measuredWidth } ?: 0
+        else -> layoutParams.width
+      }
+
+      val height = when (layoutParams.height) {
+        LayoutParams.MATCH_PARENT -> heightMeasureSpec
+        LayoutParams.WRAP_CONTENT -> widgetChildren.maxOfOrNull { it.value.measuredHeight } ?: 0
```
Comment (underscoretang) [inline redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt:77]: not sure if this is the best way to capture this, but leaving it up to `super.onMeasure` to figure this out didn't really work

File: redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt (hunk 1 above)
Comment (swankjesse) [reply redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt:83]: I don’t think we want this exact call, but there’s probably some side-effect of this call that we need.

Lemme take a look in follow-up!

**Review by swankjesse — APPROVED** [review]

**Pull request merged**